src/components/molecules/LiveSchedule/LiveSchedule.tsx

```diff
@@ -0,0 +1,86 @@
+import { useSelector } from "hooks/useSelector";
+import { useUser } from "hooks/useUser";
+import React, { FC } from "react";
+import { VenueEvent } from "types/VenueEvent";
+import { isEventLive } from "utils/event";
+import { currentTimeInUnixEpoch, formatHourAndMinute } from "utils/time";
+import { isExternalUrl } from "utils/url";
+import { enterRoom } from "utils/useLocationUpdateEffect";
+import "./LiveSchedule.scss";
+
+const LiveSchedule: FC = () => {
+  const { user, profile } = useUser();
+  const { venueEvents, venue } = useSelector((state) => ({
+    venueEvents: state.firestore.ordered.venueEvents,
+    venue: state.firestore.ordered.currentVenue[0],
+  }));
+
+  const liveEvents =
+    venueEvents && venueEvents.length
+      ? venueEvents.filter((event) => isEventLive(event))
+      : [];
+
+  const onEventClick = (event: VenueEvent) => {
+    const room = venue?.rooms?.find((room) => room.title === event.room);
+
+    const isExternal = isExternalUrl(room.url);
+    if (isExternal) {
+      window.open(room.url, "_blank", "noopener,noreferrer");
+    } else {
+      window.location.href = room.url;
+    }
+    enterRoom(
+      user!,
+      { [`${venue.name}/${room.title}`]: currentTimeInUnixEpoch },
+      profile?.lastSeenIn
+    );
+  };
+
+  return (
+    <div className="schedule-container show">
+      {!!liveEvents.length && (
```

Comment:
as above RE: using multiple `return`s to avoid nesting in JSX

Reply:
Imo, the multiple returns is not a great convention, it looks weird, it breaks react's composition, and if you have more than 1 case it's just a long list of if statements and sometimes you need to scroll back and forth.

I think something like this:
```jsx
return (
<Fragment>
 {!condition1 && <Component1/>}
 {condition1 && <Component2/>}
 {!condition2 && <Component3/>}
 {condition2 && <Component4/>}
</Fragment>
)
```

is much more readable than this:

```jsx

if (!condition1) {
 return <Component1/>
)

if (condition1) {
 return <Component2/>
)

if (!condition2) {
 return <Component3/>
)

return <Component4/>
```

But again, this is just my opinion and preference.

Reply:
For short cases like your example, I would definitely agree that your preferred method looks cleaner.

Where I find my preferred more useful is in more complex examples (eg. long chunks of JSX), particularly where it may be longer than a screen length (at which point it takes extra cognitive effort to figure out what the nesting structure/etc is, and which part it falls into. An alternative solution in that cause may be that that component is too big, and should be broken up into smaller sub-components.

This is NB/nit/stylistic, and we don't have a clearly established pattern of when/how/why to do it in the codebase + haven't discussed the pro's/con's of that as a team yet, so happy for you to leave this as is here.

Reply:
fixed in 70e8253242c4c56f3678a5035e0e6566912da044